catalog/plugin/catalog-plugin-metadatabackup/src/main/java/org/codice/ddf/catalog/plugin/metadata/backup/MetadataBackupPlugin.java

```diff
@@ -0,0 +1,313 @@
+
+    private static final Integer MAX_FOLDER_DEPTH = 4;
+
+    private static final Integer MIN_FOLDER_DEPTH = 0;
+
+    private Boolean keepDeletedMetacards = false;
+
+    private String metacardTransformerId;
+
+    private MetacardTransformer metacardTransformer;
+
+    private String outputDirectory;
+
+    private Integer folderDepth;
+
+    @Override
+    public ProcessRequest<ProcessCreateItem> processCreate(
+            ProcessRequest<ProcessCreateItem> processRequest) throws PluginExecutionException {
+        processRequest(processRequest);
+        return processRequest;
+    }
+
+    @Override
+    public ProcessRequest<ProcessUpdateItem> processUpdate(
+            ProcessRequest<ProcessUpdateItem> processRequest) throws PluginExecutionException {
+        processRequest(processRequest);
+        return processRequest;
+    }
+
+    @Override
+    public ProcessRequest<ProcessDeleteItem> processDelete(
+            ProcessRequest<ProcessDeleteItem> processRequest) throws PluginExecutionException {
+        if (keepDeletedMetacards) {
+            return processRequest;
+        }
+
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new PluginExecutionException(
+                    "Unable to delete backup ingested metadata; no output directory specified.");
+        }
+
+        List<ProcessDeleteItem> processUpdateItems = processRequest.getProcessItems();
+        for (ProcessDeleteItem processUpdateItem : processUpdateItems) {
+            Metacard metacard = processUpdateItem.getMetacard();
+            deleteMetadataIfPresent(metacard.getId());
+        }
+
+        return processRequest;
+    }
+
+    public void setOutputDirectory(String outputDirectory) {
+        this.outputDirectory = outputDirectory;
+    }
+
+    public String getOutputDirectory() {
+        return outputDirectory;
+    }
+
+    public void setFolderDepth(Integer folderDepth) {
+        folderDepth = Math.max(folderDepth, MIN_FOLDER_DEPTH);
+        folderDepth = Math.min(folderDepth, MAX_FOLDER_DEPTH);
+        this.folderDepth = folderDepth;
+    }
+
+    public Integer getFolderDepth() {
+        return folderDepth;
+    }
+
+    public void setKeepDeletedMetacards(Boolean keepDeletedMetacards) {
+        this.keepDeletedMetacards = keepDeletedMetacards;
+    }
+
+    public Boolean getKeepDeletedMetacards() {
+        return keepDeletedMetacards;
+    }
+
+    public void setMetacardTransformerId(String metacardTransformerId) {
+        this.metacardTransformerId = metacardTransformerId;
+        this.metacardTransformer = lookupTransformerReference();
+    }
+
+    public void setMetacardTransformer(MetacardTransformer metacardTransformer) {
+        this.metacardTransformer = metacardTransformer;
+    }
+
+    public String getMetacardTransformerId() {
+        return metacardTransformerId;
+    }
+
+    public void refresh(Map<String, Object> properties) {
```

Comment:
Unfortunately, I haven't seen a cleaner way to do this.  We need to type check the parameters that come in from the properties map.